src/chain/mod.rs

```diff
@@ -98,3 +80,41 @@ pub async fn execute(port: u16) -> Result<()> {
 
     Ok(())
 }
+
+async fn wait_for_anvil(port: u16, max_attempts: u16) -> Result<()> {
```

Comment:
execute() should be last function in file

in general the main entrypoint should either be first or last so its easy to find. its arbitrary which, but in kit we do last.